src/OrbitGl/Batcher.cpp

```diff
@@ -220,15 +253,33 @@ void Batcher::AddTriangle(const Triangle& triangle, const Color& color,
 
 void Batcher::AddTriangle(const Triangle& triangle, const Color& color, const Color& picking_color,
                           std::unique_ptr<PickingUserData> user_data) {
+  std::array<Color, 3> colors;
+  colors[0] = color;
+  colors[1] = color;
+  colors[2] = color;
+  AddTriangle(triangle, colors, picking_color, std::move(user_data));
+}
+
+void Batcher::AddShadedTriangle(const Triangle& triangle, const Color& color,
+                                std::unique_ptr<PickingUserData> user_data,
+                                ShadingDirection shading_direction) {
+  std::array<Color, 3> colors;
+  GetTriangleGradientColors(color, &colors, shading_direction);
```

Comment:
This assumes a certain ordering of the vertices of the triangle, we should probably have a util function that creates a triangles with the proper ordering of vertices (CreateTriangle(Vec3 bottom_left, Vec3 top_left, Vec3  top_right);). Or even something like CreateTriangleFromBox( Box, TriangleId ) where TriangleId would represent which of the 4 possible triangles to pick.

Reply:
I changed the code a bit. We now have AddShadedTrapezium instead of AddShadedTriangle. That gets the for coordinates and simply re-uses the existing GetBoxGradientColors method.

Reply:
> Hi Florian, have you considered something like this:
> 
> ![image](https://user-images.githubusercontent.com/3687222/106977042-682e4d00-670e-11eb-8f2c-bd0392322710.png)
> 
> It seems like it would simplify a lot of the logic and leave more room for text.

I really like your approach Pierric. In my opinion it seems to be more intuitive and easy to understand. Actually I tried to draw something to recommend it (your image is clearly nicer :) ). But what to do when we have 3 or more events which intersects each other? I wasn't able to draw something nice and simpler to understand, and the trapezium way seems to be clearly straightforward. I propose to have it as a possible future improve, after receiving some feedback from developers. What do you think?

<img width="491" alt="3-intersecting events" src="https://user-images.githubusercontent.com/8610429/107235384-ea778380-6a24-11eb-9db4-da61803ae193.png">

Reply:
Actually I feel like I don't have enough information about the data we are trying to draw. It seems to me that it would be possible to have a scope completely embedded in another one, which would require a different approach. I have a meeting planned with Florian tomorrow, hopefully it will clear things up. Thanks for the drawings btw.